digamma.go

```diff
@@ -0,0 +1,26 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mathext
+
+import (
+	"math"
+)
+
+// Digamma returns the logorithmic derivative of the gamma function at x.
+//  ψ(x) = d/dx (Ln (Γ(x)).
+func Digamma(x float64) float64 {
+	// This is adapted from
```

Comment:
Wolfram lists the results as complex infinity, this returns negative inf.  I have a feeling it depends on the method of approximation, wolfram looks right but there isn't a way to represent it in float64.

An alternative implementation could use the Chebychev approximations in http://www.ams.org/journals/mcom/1961-15-074/S0025-5718-61-99221-3/S0025-5718-61-99221-3.pdf